api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -113,6 +113,16 @@ default Integer sortOrderId() {
     return null;
   }
 
+  /** Returns the data sequence number of the snapshot in which the file should be applied. */
+  default Long dataSequenceNumber() {
+    return null;
+  }
+
+  /** Returns the file sequence number. */
```

Comment:
We, probably, need to add more context on what this sequence number represents given that it is a public API. What about something like this?

```
/**
 * Returns the file sequence number.
 *
 * <p>The file sequence number represents the sequence number of the snapshot in which the
 * underlying file was added. The file sequence number is always assigned at commit and cannot be
 * provided explicitly, unlike the data sequence number. The file sequence number does not change
 * upon assigning.
 *
 * <p>This method can return null if the file sequence number is unknown. This may happen while
 * reading a v2 manifest that did not persist the file sequence number for manifest entries with
 * status EXISTING or DELETED (older Iceberg versions).
 */
```

Reply:
> The file sequence number is always assigned at commit and cannot be  provided explicitly, unlike the data sequence number. The file sequence number does not change upon assigning.

Can we also use the compaction example? 
```
The file sequence number is always assigned at commit. In case of rewrite (like compaction), file sequence number can be higher than the data sequence number.
```

Reply:
Adding the compaction example makes sense to me.

Reply:
Sure, thanks for the suggestion. The comments I added to ContentFile are indeed a bit light. My thinking was that I didn't want to duplicate the comments from ManifestEntry, but you're right, it makes sense to be verbose with the comments in ContentFile as well.
I merge the 2 proposals above and added to the function.